### PR TITLE
Ab#8296 site brand support

### PR DESCRIPTION
**Pull request by PeculiarGoat**
ADO card: ☑️ [AB#8296](https://dev.azure.com/S72/SHIFT72/_workitems/edit/8296)

- [x] Has this been discussed with Delivery Team?

## Description of work
This is a bit of work to pre-emt the next Kibble release containing https://github.com/shift72/kibble/pull/127.
- Modified application.jet to use self-service CSS files via the site model.
- Modified various jets to use self-service brand images (logo@1x, logo@2x, favicon, icons-192, icons-512, facebook-image, sponsors-md, sponsors-xs, sponsors-xxxs) via the site model.
- Tidied up some of the image related CSS.
- Replaced the 5 navbar-brand variables with a single clearer variable.  Added a matching footer-brand variable because why not.

## Caveats
- App logos can be uploaded to the site brand API but Kibble doesn't copy the file into the site build output like it does with the email logo.  For now, we can add a step to the provisioning/styling process to manually add the correct app logo URL to the client brand table in Uber.  I will create a task to look into automating this (I reckon the site brand API should throw the URL in the client brand table for us).

## Config settings/Toggles required for the feature to work
- `self_service_site_images` for the images.
- `self_service_css` for the CSS.
- NOTE: If a CSS file or any of the image files don't have database entries, the template will fallback onto default i.e. no custom CSS file, and whichever image from core-template.

### Any PRs which this PR depends on
- https://github.com/shift72/kibble/pull/127

### Affected Clients
 - None until we do it live.

## Checklist
- [x] CI tests are passing Github actions (inc. linting)
- [x] Key areas of the feature outlined for context and testing
- [x] Have checked this at multiple screen resolutions and range of browsers
- [x] Moved ADO card to Dev/done, checked link to Github, tagged "Review"
- [x] Updated changelog (if applicable)
- [x] I promise to document any new feature toggles/configurations in the appropriate documentation

### CODE REVIEW

File: site/templates/application/application.jet [hunk 1]
```diff
@@ -7,11 +7,8 @@
 {{CDN := "//cdn.shift72.com/latest"}}
 {*{CDN := "//localhost:3000"}*}
 
-{{CSSFilename := ""}}
-{{if isEnabled("self_service_css") && site.CSSFilename != "" }}
-  {{CSSFilename = site.CSSFilename}}
-{{end}}
-{*{CSSFilename = "local.css"}*}
+{{CSSFileURL := site.SiteBrand.GetLink("css", "")}}
+{*{CSSFileURL = "local.css"}*}
```
Comment (additiverich) [inline site/templates/application/application.jet:11]: Maybe make this a `:=` type assignment (sorry don't know the words)

**Comment by PeculiarGoat** [comment]
@additiverich @matthewthomsonnz @l0ud0gg bumped kibble to 0.17.0... unblocked and ready for review.

**Comment by l0ud0gg** [comment]
we ran it up this afternoon and managed to get three sites running off one template branding differently;

- https://test2.shift72.com/ (has uploaded site images and css)
- https://test3.shift72.com/ (has uploaded site images and css)
- https://test-template.shift72.com/ (does not have any custom images or css, so using fallbacks)

sites working as expected for each of the image types. The only thing i havent checked yet is the sponsor banners so will come back to that and report findings.

**Comment by matthewthomsonnz** [comment]
This seems fine, are there any docs on how to use the self service api and add the image/css database entries?

**Comment by PeculiarGoat** [comment]
@matthewthomsonnz 
```
curl -i -v \
-X PUT \
-H 'x-auth-token: 76ea78f3cf3f8efce2fee9e95efc42c1' \
"https://core-store.shift72.local/services/users/v1/css" \
-F 'file=@/home/jack/custom.css'
```

```
curl -i 'https://staging-store-kibble.shift72.com/services/users/v1/site_brand/image' \
  -H 'X-AUTH-TOKEN: ffe371807d90164d0e31c047cf3db44f' \
  -F "file=@app-logo.png" \
  -F image="{\"type\":\"app-logo\"}" \
  -H 'Accept: application/json, text/plain, */*'
```

Valid image types are:
```
    VALID_IMAGE_TYPES = [
      "logo@1x",
      "logo@2x",
      "favicon",
      "icons-192",
      "icons-512",
      "facebook-image",
      "app-logo",
      "email-logo",
      "sponsors-md",
      "sponsors-xs",
      "sponsors-xxxs"
    ]
```

**Review by matthewthomsonnz — APPROVED** [review]
The site image stuff is good nice work.

Have to take the css part on faith. I ran into issues with self service css on multiple levels, when starting the template pointed to docker dev. Would be good to make that local dev experience better at some stage if possible

**Comment by PeculiarGoat** [comment]
@matthewthomsonnz what were the issues with CSS stuff?

**Comment by matthewthomsonnz** [comment]
> @matthewthomsonnz what were the issues with CSS stuff?

@PeculiarGoat here's a rough writeup from memory of my experience this morning. Thinking perhaps we could do a huddle to make a some adjustments:

- (posting css) API doesnt handle if file param is not a file (e.g. string)
- Had to add Site entry to db so it can GET /services/users/v1/css without 422
- Had to change  `const filename = readJsonValueFromRawData('css_filename', d);` to:  `const filename = readJsonValueFromRawData('css_path', d)` because where is it getting css_filename from when it returns this: https://github.com/indiereign/shift72-users/blob/master/ruby/api/v1/css.rb#L40
- had to change `const https = require('https');` to `const https = require('http');`
- had to figure out what kibble was trying to do because returning an invalid css path without domain name i think
- had to figure out how to make site build worker serve css file somehow

There were probably a couple other things but thats the jist haha

**Comment by PeculiarGoat** [comment]
@matthewthomsonnz oh I forgot those scripts were there lol they do need updating... the current 'dev experience' is just curl.

Sounds like the rest of your issues are due to pointing it at docker-dev though... not sure why you would do that?

**Comment by PeculiarGoat** [comment]
@matthewthomsonnz I've fixed up those scripts... they are working fine against staging sites (note: they require `self_service_css` feature toggle to be on).  Not sure how well they work with docker-dev but meh.

**Pull request merged**